judge/admin/submission.py

```diff
@@ -265,3 +267,14 @@ def judge_view(self, request, id):
             raise PermissionDenied()
         submission.judge(rejudge=True)
         return HttpResponseRedirect(request.META.get('HTTP_REFERER', '/'))
+
+    def disqualify_view(self, request, id):
+        if not request.user.has_perm('judge.disqualify_submission'):
+            raise PermissionDenied()
+        submission = get_object_or_404(Submission, id=id)
+        if submission.contest_or_none is None:
+            raise Http404()
+        submission.contest.is_disqualified = not submission.contest.is_disqualified
+        submission.contest.save()
+        submission.contest.participation.recompute_results()
```

Comment:
We need to decide - do we want to recompute the results for the *entire* participation if the scoreboard is supposed to be frozen? This could have potentially unwanted side effects.

Reply:
I say no, however I think you suggested to do it this way.

Reply:
Actually, ignore this. It turns out frozen scoreboard stops at the contest submission level, not the contest participation level. This means that recomputing participation restuls is fine.